app/javascript/lib/plot.js

```diff
@@ -17,6 +17,22 @@ const PlotUtils = function() {
 // special legend entries with custom sorting rules
 const SPECIAL_LEGEND_ENTRIES = [UNSPECIFIED_ANNOTATION_NAME, FILTERED_TRACE_NAME]
 
+/**
+ * Enables handling labels with commas in legend filtering.
+ * Used for non-display comparison, not for rendering text.
+ *
+ * Example:
+ * central memory CD4-positive, alpha-beta T cell
+ */
+export function safenLabels(labels) {
```

Comment:
Just out of curiosity, why did you opt for the straight replacement instead of a built-in function like `encodeURIComponent`?  The latter seems to have been our stock answer for dealing with special characters in URLs.

Reply:
Upon further manual testing, I see simply applying `encodeURIComponent` breaks legend filtering when there are multiple entries in the query.  So I've reverted my most recent commit; now comma is replaced with hyphen like before.